### DISCUSSION TITLE
Question: Using the Tree component, why it doesn't work with tap, pinch, and swipe touchscreen events?

### DISCUSSION THREAD

**Discussion by desioncezar** [primefaces/community#2377] [PrimeReact]
Hi everybody, I hope you are fine. You are doing excellent work on Primereact. I love it. :)

We started developing a React App with some functionalities to drag and drop, but the touchscreen cannot work with the Tree component.

We realized that the problem is the Tree component that doesn't handle these tap, pinch, and swipe touchscreen events.

![Screenshot 2024-07-24 085914](https://github.com/user-attachments/assets/1319f535-c9b4-486f-896e-e705802cb13e)

Code:
 
                   
                   <Tree
                      value={dataContext.profileMenu}
                      selectionMode={dataContext.profileMenuSelectionMode}
                      draggableNodes={true}
                      dragdropScope='demo'
                      onDragDrop={(e) => onDragDrop(e)}
                      selectionKeys={dataContext.profileMenuSelected}
                      expandedKeys={expandedKeys}
                      onToggle={(e) => setExpandedKeys(e.value)}
                      onUnselect={() => dataContext.setProfileMenuSelected(null)}
                      onSelectionChange={(e) => onSelected(e)}
                      filter
                      filterMode="lenient"
                      className="border-noround"
                      style={{ border: 0, background: '#ffffff' }}
                      filterPlaceholder="Filter"
                  />

These components below work except for the tap, and double tap events.
https://mdbootstrap.com/docs/react/methods/touch/

How to enable touchscreen events on the Tree component?

Thank you

**Comment by melloware** [comment]
Not all components have been tested for touch friendliness. I am sure it's just a missing feature.
